controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -1045,6 +1055,34 @@ func (r *ReconcileHyperConverged) detectTaintedConfiguration(req *common.HcoRequ
 	}
 }
 
+func (r *ReconcileHyperConverged) getSystemHealthStatus(req *common.HcoRequest) string {
+	isConditionAvailableTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionAvailable)
+	isConditionDegradedTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionDegraded)
+	isConditionProgressingTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionProgressing)
+	isConditionReconcileCompleteTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionReconcileComplete)
+
+	isSystemHealthStatusError := !isConditionAvailableTrue || isConditionDegradedTrue
+	if isSystemHealthStatusError {
```

Comment:
The `isSystemHealthStatusError` variable is only used here. I think we can drop it, and to use the condition directly for the if; e.g.
```suggestion
	if !isConditionAvailableTrue || isConditionDegradedTrue {
```

Another alternative is to write "micro-functions" so we'll preserve the readability; e.g.
```golang
func isSystemHealthStatusError(conditions HcoConditions) {
    return !req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionAvailable) || req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionDegraded)
}

...
    if isSystemHealthStatusError(req.Conditions) { 
        ...
```

What do you think?

Reply:
Took the second approach. thanks